### DISCUSSION TITLE
How to achieve HA for code location service?

### DISCUSSION THREAD

**Discussion by shuoli84** [dagster-io/dagster#18699] [Q&A] [answered]
Currently we encounter GRPC unavailable quite a bit, and the default deployment replica is 1. Can I just increase it to like 4 to spread the load? Thanks~

**Comment by baowei** [comment]
I also encountered this problem when deploying EKS. I want to confirm what impact will it have if the Code location server replica is set greater than 2?
As described in the documentation: “Code location servers serve metadata about the collection of its Dagster definitions. You can have many code location  servers, but currently each code location can only have one replica for its server.”

**Comment by shalabhc** [comment, marked as answer]
Multiple GRPC code server replicas for a single agent are currently not supported but this feature is on our roadmap. You can however run multiple replicas of the agent and each agent will spin up its own set of code servers. See https://docs.dagster.io/dagster-cloud/deployment/agents/running-multiple-agents.

**Comment by shalabhc** [comment]
Do you know why the GRPC server becomes unavailable? Do you see high CPU or memory utilization on these servers? It may be worthwhile increasing the capacity of the code servers to make it more reliable.

**Comment by shuoli84** [comment]
@shalabhc thanks for the reply. 
It seems the agent is only for cloud deployment? 
I am investigating the unavailable issue, but just redeployment can cause a short period of time of unavailability I think? 
BTW, I am running the on-premise version, not dagster cloud.
